README.rst

```diff
@@ -20,7 +20,7 @@ Geographic information systems use GeoTIFF and other formats to organize and
 store gridded, or raster, datasets. Rasterio reads and writes these formats and
 provides a Python API based on N-D arrays.
 
-Rasterio 1.4 works with Python 3.9+, Numpy 1.21+, and GDAL 3.3+. Official
+Rasterio 1.4 works with Python 3.10+, Numpy 1.24+, and GDAL 3.5+. Official
```

Comment:
@snowman2 apparently there are Fiona and Rasterio users who are still on 3.8. How about we split the difference and keep 3.9?